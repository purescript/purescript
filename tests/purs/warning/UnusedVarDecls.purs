-- @shouldWarnWith UnusedName
-- @shouldWarnWith UnusedDeclaration

module Main
  ( unusedArgDecl
  , X(..)
  ) where

data X = X


unusedArgDecl :: X -> X
unusedArgDecl unusedArg = X

unusedDecl :: X
unusedDecl = 
  X