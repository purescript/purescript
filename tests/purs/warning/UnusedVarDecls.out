Warning 1 of 2:

  in module [33mMain[0m
  at tests/purs/warning/UnusedVarDecls.purs:13:15 - 13:24 (line 13, column 15 - line 13, column 24)

    Name [33munusedArg[0m was introduced but not used.

  in value declaration [33munusedArgDecl[0m

  See https://github.com/purescript/documentation/blob/master/errors/UnusedName.md for more information,
  or to contribute content related to this warning.

Warning 2 of 2:

  in module [33mMain[0m
  at tests/purs/warning/UnusedVarDecls.purs:16:1 - 17:4 (line 16, column 1 - line 17, column 4)

    Declaration [33munusedDecl[0m was not used, and is not exported.


  See https://github.com/purescript/documentation/blob/master/errors/UnusedDeclaration.md for more information,
  or to contribute content related to this warning.

